src/notebooks/controllers/controllerLoader.ts

```diff
@@ -74,12 +64,25 @@ export class ControllerLoader implements IControllerLoader, IExtensionSyncActiva
         // Make sure to reload whenever we do something that changes state
         const forceLoad = () => this.loadControllers(true);
         this.serverUriStorage.onDidChangeUri(forceLoad, this, this.disposables);
-        this.kernelProvider.onDidStartKernel(forceLoad, this, this.disposables);
+
+        // If we create a new kernel, we need to refresh if the kernel is remote (because
+        // we have live sessions possible)
+        // Note, this is a perf optimization for right now. We should not need
+        // to check for remote if the future when we support live sessions on local
+        this.kernelProvider.onDidStartKernel((k) => {
+            if (isRemoteConnection(k.kernelConnectionMetadata)) {
+                forceLoad().ignoreErrors();
```

Comment:
the ignoreErrors here looks like it might be a handling change from the previous code that might not be intended. From the comment on line 112 might be errors getting thrown by loadControllers that now won't bubble up.

Reply:
Although all of the handlers are that way. Maybe it would be better to make forceLoad at least log the error and then have it not return a promise.